feat: add more tools for developer experience

anriuss opened the pull request:
Added husky to use pre-commit hooks in order to prevent unwanted bugs on production.

Fixed error with type of any.

Added output for prisma

anriuss committed:
feat: add more tools for developer experience

anriuss committed:
fix: eslint error

anriuss committed:
fix: eslint error

anriuss committed:
chore: update README.md

RomaLetodiani committed:
chore: add husky dependency to package-lock.json

- Introduced husky version 9.1.7 to enhance Git hooks management, improving the development workflow.
- This addition supports better automation and ensures code quality through pre-commit and pre-push hooks.

RomaLetodiani committed:
chore: update pre-commit hook to support bun or npm

- Enhanced the pre-commit script to check for the presence of 'bun' and fall back to 'npm' if not installed, improving flexibility in running pre-commit checks.
- This change ensures that formatting, linting, type checking, and building processes are executed consistently, regardless of the package manager used, thereby streamlining the development workflow.

RomaLetodiani commented on .husky/pre-commit:
```diff
@@ -0,0 +1,56 @@
+
+# Run type check
+echo "📝 Running type check..."
+bun run typecheck || {
```
We should delegate heavier checks to CI/CD.

I will move tasks like type-checking and building to the CI/CD pipeline. Developers can run these tasks locally when needed (e.g., via npm run check).

RomaLetodiani committed:
chore: integrate lint-staged for pre-commit checks and update dependencies

- Added lint-staged to run formatting and linting on staged files before commits, enhancing code quality and consistency.
- Updated the pre-commit hook to utilize lint-staged, streamlining the pre-commit process by removing redundant checks and focusing on staged files.
- Updated package.json and package-lock.json to include lint-staged and its dependencies, ensuring proper integration into the development workflow.

RomaLetodiani committed:
chore: enhance pre-commit hook with informative output messages

- Added informative output messages to indicate the status of lint-staged checks, enhancing user feedback during the commit process.

RomaLetodiani commented on .husky/pre-commit:
What do you think about using a "lint-staged" library instead?

RomaLetodiani reviewed (commented):
Thank you for your contributions! Please review the newly added commits and share your feedback.
